lib/client/tool_utils.go

```diff
@@ -0,0 +1,134 @@
+/*
+Copyright 2016-2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package client
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"strings"
+	"time"
+
+	"github.com/gravitational/teleport"
+	"github.com/gravitational/teleport/api/constants"
+	"github.com/gravitational/teleport/api/types/events"
+	apievents "github.com/gravitational/teleport/api/types/events"
+	"github.com/gravitational/teleport/lib/asciitable"
+	"github.com/gravitational/teleport/lib/utils"
+	"github.com/gravitational/trace"
+)
+
+// DefaultFormats is the default set of formats to use for commands that have the --format flag.
+var DefaultFormats = []string{teleport.Text, teleport.JSON, teleport.YAML}
+
+// FormatFlagDescription creates the description for the --format flag.
+func FormatFlagDescription(formats ...string) string {
+	return fmt.Sprintf("Format output (%s)", strings.Join(formats, ", "))
+}
+
+func DefaultSearchSessionRange(fromUTC, toUTC string) (from time.Time, to time.Time, err error) {
+	from = time.Now().Add(time.Hour * -24)
+	to = time.Now()
+	if fromUTC != "" {
+		from, err = time.Parse(time.RFC3339, fromUTC)
+		if err != nil {
+			return time.Time{}, time.Time{},
+				trace.BadParameter("failed to parse session listing start time: expected format %s, got %s.", time.RFC3339, fromUTC)
+		}
+	}
+	if toUTC != "" {
+		to, err = time.Parse(time.RFC3339, toUTC)
+		if err != nil {
+			return time.Time{}, time.Time{},
+				trace.BadParameter("failed to parse session listing end time: expected format %s, got %s.", time.RFC3339, toUTC)
+		}
+	}
+	return from, to, nil
+}
+
+// GetPaginatedSessions wraps the given sessionGetter function with the required logic to use it's pagination. Returns up to 'max' sessions.
+func GetPaginatedSessions(max int, sessionGetter func(startKey string) ([]apievents.AuditEvent, string, error)) ([]apievents.AuditEvent, error) {
+	prevEventKey := ""
+	sessions := []apievents.AuditEvent{}
+	for {
+		nextEvents, eventKey, err := sessionGetter(prevEventKey)
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+		sessions = append(sessions, nextEvents...)
+		if eventKey == "" || len(sessions) > max {
+			break
+		}
+		prevEventKey = eventKey
+	}
+	if max < len(sessions) {
+		return sessions[:max], nil
+	}
+	return sessions, nil
+}
+
+type SessionsCollection struct {
```

Comment:
I'm not sure this belongs in lib/client TBH. Can we make a "common" subpackage in tool and put it there instead?

Reply:
This was already suggested, he made the change, then I suggested the opposite https://github.com/gravitational/teleport/pull/12763#discussion_r916110339 🤦‍♂️ Sorry Edward. I'll defer to Roman on this one